src/encoded/static/components/viz/Workflow/EdgesLayer.js

```diff
@@ -5,11 +5,242 @@ import PropTypes from 'prop-types';
+            return bestSegment;
+        }
+
+        const originalEdgesSortedByLength = originalEdges.slice(0).sort(function(edgeA, edgeB){
+            // Handle the shorter edges first
+            const { source: sA, target: tA } = edgeA;
+            const { source: sB, target: tB } = edgeB;
+            const xDistA = Math.abs(tA.x - sA.x);
+            const xDistB = Math.abs(tB.x - sB.x);
+
+            if (xDistA < xDistB) return -1;
+            if (xDistA > xDistB) return 1;
+
+            const yDistA = Math.abs(tA.y - sA.y);
+            const yDistB = Math.abs(tB.y - sB.y);
+
+            if (yDistA < yDistB) return -1;
+            if (yDistA > yDistB) return 1;
+
+
+
+            return 0;
+        });
+
+        return originalEdgesSortedByLength.map(function(edge, edgeIdx){
+            const { source, target } = edge;
+            const { column: sourceCol, x: sourceX, y: sourceY } = source;
+            const { column: targetCol, x: targetX, y: targetY } = target;
+            const columnDiff = targetCol - sourceCol;
+            if (columnDiff <= 0){
+                // Shouldn't happen I don't think except if file is re-used/generated or some other unexpected condition.
+                console.error("Target column is greater than source column", source, target);
+                return edge; // Skip tracing it.
+            }
+            if (columnDiff === 1){
+                return edge; // Doesn't need to go around obstacles, skip.
+            }
+
+            const vertices = [[ sourceX + columnWidth, sourceY ]];
+
+            let prevY = sourceY;
+            for (let colIdx = sourceCol + 1; colIdx < targetCol; colIdx++){
+                //const yDiff = targetY - prevY;
+                //const idealYCoord = prevY + (yDiff / 2); // (((colIdx - sourceCol) / columnDiff) * yDiff);
+                const bestSegment = getNearestSegment(
+                    colIdx,
+                    prevY,
+                    targetY,
+                    originalEdgesSortedByLength.slice(0, edgeIdx)
+                );
+                if (!bestSegment){
+                    throw new Error("Could not find viable path for edge");
+                }
+                const [ [ bsX, bsY ], [ beX, beY ] ] = bestSegment;
+                vertices.push([ bsX - nodeEdgeLedgeWidths[0], bsY ]);
+                vertices.push([ beX + nodeEdgeLedgeWidths[1], beY ]);
+                prevY = beY;
+            }
+            vertices.push([ targetX, targetY ]);
+
+            return _.extend({ vertices }, edge);
+        });
+
+    }
+
+    let horizontalSegments;
+    let tracedEdges = null;
+    let attempts = 0;
+    while (!tracedEdges && attempts < 5){
+        horizontalSegments = buildVisibilityGraph(4 + attempts);
+        try {
+            tracedEdges = assembleSegments(horizontalSegments, 4 + attempts);
+        } catch (e){
+            tracedEdges = null;
+            if (e.message === "Could not find viable path for edge"){
```

Comment:
Will include `|| attempts === 4` in if condition & remove from while loop condition